drv/stm32h7-qspi/src/lib.rs

```diff
@@ -258,8 +316,24 @@ impl Qspi {
         // perform transfers.
         let mut out = out;
         while !out.is_empty() {
+            let sr = self.reg.sr.read();
+
+            if sr.tof().bit_is_set() {
+                self.reg.fcr.modify(|_, w| w.ctof().set_bit());
+                self.disable_all_interrupts();
+                return Err(QspiError::Timeout);
+            }
+            if sr.tef().bit_is_set() {
+                self.reg.fcr.modify(|_, w| w.ctef().set_bit());
+                self.disable_all_interrupts();
```

Comment:
similarly, is it worth refactoring this so that the error path disables interrupts in one place instead of multiple?

Reply:
I'll give this a shot (and I deeply appreciate you having an eye for things like this!)